lib/ngannotate/processor.rb

```diff
@@ -29,10 +29,14 @@ def force
     # Skip processing in environments where it does not make sense.
     # Override by NG_FORCE=true env variable
     #
-    def skip
+    def skip_by_config
       !force && !::Rails.configuration.ng_annotate.process
     end
 
+    def skip
+      skip_by_config || !@file.start_with?(::Rails.root.to_s)
```

Comment:
@see http://guides.rubyonrails.org/engines.html

For example,
https://github.com/kikonen/gi_album_engine mounted into https://github.com/kikonen/host. Engine has various assets, but host app takes care of assets precompile in deployment. And naturally assets owned by engine are not under rails root dir, but within "gems" directory.